crates/debug/src/transform/expression.rs

```diff
@@ -604,6 +602,22 @@ where
                     // Don't re-enter the loop here (i.e. continue), because the
                     // DW_OP_deref still needs to be kept.
                 }
+                Operation::Shr { .. } | Operation::Shra { .. } => {
+                    // Insert value normalisation part.
+                    let mut writer = ExpressionWriter::new();
+                    writer.write_op(gimli::constants::DW_OP_swap)?;
+                    writer.write_op(gimli::constants::DW_OP_const1u)?;
+                    writer.write_u8(32)?;
+                    writer.write_op(gimli::constants::DW_OP_shl)?;
+                    writer.write_op(gimli::constants::DW_OP_const1u)?;
+                    writer.write_u8(32)?;
+                    // Extend with the (arithmetic) shift.
+                    writer.write_u8(buf[pos])?;
+                    writer.write_op(gimli::constants::DW_OP_swap)?;
```

Comment:
How about `DW_OP_swap DW_OP_plus_uconst 32` instead of `DW_OP_const1u 32 shr* DW_OP_swap`?

Reply:
I wanted to present it as clearly as possible in the first commit, for posterity :-)
I am preparing a commit, that adds an optimisation.

Reply:
Mostly because I was not sure whether two shifts can be collapsed to a single one by the DWARF spec. But I now assume this is the case.

Reply:
This is now fe7f041.